runtime/src/main/java/org/corfudb/security/tls/ReloadableTrustManager.java

```diff
@@ -40,16 +40,33 @@ public ReloadableTrustManager(TrustStoreConfig trustStoreConfig) {
 
     @Override
     public void checkClientTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+        if (watcher.isCertExpiryCheckDisabled()) {
```

Comment:
This change  completely bypasses cert check if the expiry check is disabled.
We don't want that. We want to still verify that the cert is trusted, just ignore the expiry.

Something like this:

public void checkClientTrusted(X509Certificate[] chain, String authType) throws CertificateException {
      if  (!watcher.isCertExpiryCheckDisabled()) {
           for (X509Certificate cert : chain) {
                cert.checkValidity();
           }
       } 

        X509TrustManager trustManager = getTrustManager();
        trustManager.checkClientTrusted(chain, authType);
}